### ISSUE TITLE
DuplicateMapOrSetKeyCheck — new static check for duplicate keys/elements in Map.of, Set.of, and Map.ofEntries

### ISSUE THREAD

**Issue by AseemPrasad** [checkstyle/checkstyle#21672]
### Description
Java 9+ static factory methods like `Map.of(...)`, `Set.of(...)`, and `Map.ofEntries(...)` throw an `IllegalArgumentException` at runtime if passed duplicate keys or duplicate set elements. Checkstyle currently does not provide a check to detect duplicate key or element arguments passed to these factory methods at static analysis time.
### Proposed Check
`DuplicateMapOrSetKeyCheck`
The check should report key or element arguments passed to `Map.of`, `Set.of`, and `Map.ofEntries` whose AST structures match a preceding key or element argument in the same method call.
### Violation Example
```java
class Test {
    void process() {
        Map<String, Integer> map = Map.of(
            "key1", 1,
            "key2", 2,
            "key1", 3  // violation: duplicate key "key1"
        );
        Set<String> set = Set.of(
            "A",
            "B",
            "A"        // violation: duplicate element "A"
        );
        Map<String, Integer> entries = Map.ofEntries(
            Map.entry("k1", 1),
            Map.entry("k1", 2) // violation: duplicate entry key "k1"
        );
    }
}
Expected violations:

Duplicate key or element argument identical to the argument at line 4. Duplicate key or element argument identical to the argument at line 10. Duplicate key or element argument identical to the argument at line 16.

Non-Violation Example
java
class Test {
    void process() {
        Map<String, Integer> map = Map.of("a", 1, "b", 2);
        Set<String> set = Set.of("X", "Y", "Z");
    }
}
What should NOT be flagged
Non-factory method calls like custom map.of("a", "b").
Distinct keys or elements.
Expected Configuration
xml
<module name="Checker">
    <module name="TreeWalker">
        <module name="DuplicateMapOrSetKey"/>
    </module>
</module>
Implementation Notes
Subscribes to TokenTypes.METHOD_CALL.
Inspects target and method name (Map.of, Set.of, Map.ofEntries, Map.entry).
Structural AST equality (isAstEqual) is used so literal keys, enum constants, static fields, and expression keys are evaluated without noise from formatting or comments.

**Comment by romani** [comment]
I hesitant to add Check for this. Please share a prove that it is happening in real life.

**Comment by AseemPrasad** [comment]
The reason for this check is that **`javac` does not catch duplicate keys at compile-time**, but Java 9+ factory methods (`Map.of`, `Set.of`, `Map.ofEntries`) throw an immediate runtime `java.lang.IllegalArgumentException` as soon as the method is executed.

This might lead to silent runtime crashes in production when maintaining large maps or sets (such as feature flags, route mappings, or permission sets) during copy-paste or merge operations.

Because of this, major Java static analysis tools explicitly enforce this rule:
- **Google Error Prone**: [`DuplicateMapKeys`](https://errorprone.info/bugpattern/DuplicateMapKeys) ("Calls to Map.of / Map.ofEntries must not contain duplicate keys").
- **SonarQube / SonarLint**: Rule [`S6322`](https://rules.sonarsource.com/java/RSPEC-6322/) ("Collection factory methods should not be called with duplicate elements").
- **IntelliJ IDEA**: Built-in inspection *"Duplicate key in 'Map.of' call" / "Duplicate element in 'Set.of' call"*.

Since `javac` allows this code to compile cleanly, having a static check in Checkstyle allows projects to catch these guaranteed runtime crashes in CI before deployment.

Thank you !!

**Comment by romani** [comment]
Thanks a lot for detailed explanation

**Comment by AseemPrasad** [comment]
@romani i have raised the pr #21673 on this issue.. 
thank you..
